apps/server/src/shared/domain/rules/submission.rule.spec.ts

```diff
@@ -238,50 +238,147 @@ describe('SubmissionRule', () => {
 			});
 
 			describe('when action is "read"', () => {
-				describe('when user has write access', () => {
-					const setup = () => {
-						const permission = 'a' as Permission;
-						const user = buildUserWithPermission(permission);
-						const submission = submissionFactory.build();
-						// eslint-disable-next-line @typescript-eslint/no-explicit-any
-						const hasWriteAccessSpy = jest.spyOn(SubmissionRule.prototype as any, 'hasWriteAccess');
-						hasWriteAccessSpy.mockReturnValue(true);
-
-						return { user, submission, permission };
-					};
-
-					it('should return true', () => {
-						const { user, submission, permission } = setup();
+				describe('when submission is not submitted', () => {
+					describe('when user is a submitter', () => {
+						const setup = () => {
+							const permission = 'a' as Permission;
+							const user = buildUserWithPermission(permission);
+							const user2 = buildUserWithPermission(permission);
+							const course = courseFactory.build({ students: [user, user2] });
+							const task = taskFactory.build({ course, publicSubmissions: true });
+							const submission = submissionFactory.build({ task, student: user2, teamMembers: [user, user2] });
+							return { user, submission, permission };
```

Comment:
```suggestion
const submission = submissionFactory.build({ task, student: user2, teamMembers: [user, user2] });
							
return { user, submission, permission };
```

Auch bei den folgenden